src/crypto/mod.rs

```diff
@@ -86,3 +42,84 @@ pub enum CryptoError {
     #[error("{0}")]
     Io(#[from] io::Error),
 }
+
+/// An ID specifying which Crypto implementation to use.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum CryptoProviderId {
+    #[cfg(feature = "openssl")]
+    /// Use OpenSSL
+    OpenSsl,
+    #[cfg(all(feature = "openssl", feature = "sha1"))]
+    /// Use OpenSSL for most ciphers, but use sha1 crate for SHA1 hashes.
+    OpenSslWithSha1Crate,
+    #[cfg(feature = "wincrypto")]
+    /// Use Windows Cryptography APIs
+    WinCrypto,
+}
+
+impl Default for CryptoProviderId {
+    #[allow(unreachable_code)]
+    fn default() -> Self {
+        #[cfg(all(feature = "openssl", feature = "sha1"))]
+        return CryptoProviderId::OpenSslWithSha1Crate;
+        #[cfg(feature = "openssl")]
+        return CryptoProviderId::OpenSsl;
+        panic!("No default for CryptoProviderId!")
```

Comment:
don't you need WinCrypto here as well as a default I I compile with only Wincrypto?

Reply:
Not from what @algesten indicated he wanted the behavior to be based on how ureq3 was done with nativetls... we should never default to Wincrypto. Or so that's what I understood.